src/python/pants/notes/2.7.x.md

```diff
@@ -1,5 +1,49 @@
 # 2.7.x Stable Releases
 
+## 2.7.0.dev4 (Aug 14, 2021)
+
+### New Features
+
+* Add additional Google Cloud libraries to dependency inference's default module mapping ([#12553](https://github.com/pantsbuild/pants/pull/12553))
+
+* Validate that `pytest-cov` is installed when `[test].use_coverage` is set ([#12534](https://github.com/pantsbuild/pants/pull/12534))
+
+### User API Changes
+
+* Deprecate `--pants-supportdir`, rename `--engine-visualize-to`, and fix help for `--loop`. ([#12558](https://github.com/pantsbuild/pants/pull/12558))
+
+* Change the PythonProtobufMypyPlugin option scope. ([#12525](https://github.com/pantsbuild/pants/pull/12525))
+
+### Plugin API Changes
+
+* Upgrade packaging from 20.9 to 21.0 ([#12561](https://github.com/pantsbuild/pants/pull/12561))
+
+* Allow newer versions of setuptools and wheel. ([#12555](https://github.com/pantsbuild/pants/pull/12555))
+
+* Deprecate an options scope with underscores. ([#12532](https://github.com/pantsbuild/pants/pull/12532))
+
+* Remove `TestResult.skip()` in favor of `TestFieldSet.opt_out()` ([#12516](https://github.com/pantsbuild/pants/pull/12516))
+
+* Finally get rid of Optionable entirely. ([#12538](https://github.com/pantsbuild/pants/pull/12538))
+
+### Bug fixes
+
+* Fix `shfmt` and `--test-open-coverage` to handle CPU architecture `Platform` changes ([#12546](https://github.com/pantsbuild/pants/pull/12546))
+
+* Fix asdf directory lookup. ([#12528](https://github.com/pantsbuild/pants/pull/12528))
+
+* Fix shlexing of passthrough args. ([#12547](https://github.com/pantsbuild/pants/pull/12547))
+
+* pyenv should fall back to $HOME/.pyenv ([#12527](https://github.com/pantsbuild/pants/pull/12527))
+
+### Performance
```

Comment:
Hrm. I'm not too comfortable having this section without automated performance tests, but it looks like this section suggestion was added a good while ago.